app/src/main/java/com/keylesspalace/tusky/fragment/SFragment.java

```diff
@@ -305,12 +315,37 @@ protected void more(@NonNull final Status status, View view, final int position)
                     timelineCases.pin(status, !status.isPinned());
                     return true;
                 }
+                case R.id.status_mute_conversation: {
+                    timelineCases.muteConversation(status, status.getMuted() == null || !status.getMuted())
+                            .observeOn(AndroidSchedulers.mainThread())
+                            .as(autoDisposable(from(this, Lifecycle.Event.ON_DESTROY)))
+                            .subscribe();
+                    return true;
+                }
             }
             return false;
         });
         popup.show();
     }
 
+    private static boolean accountIsInMentions(AccountEntity account, Status.Mention[] mentions) {
+        if (account == null) {
+            return false;
+        }
+
+        for (Status.Mention mention : mentions) {
+            try {
+                if (account.getUsername().equals(mention.getUsername()) &&
+                        account.getDomain().equals(Uri.parse(mention.getUrl()).getHost())) {
+                    return true;
+                }
+            } catch (Exception exception){
+                // This block intentionally left blank
```

Comment:
What kind of exceptions can be thrown? It would be better to catch more specific ones and log an error message.

Reply:
Hm, I had thought it threw an exception on parse failure, but it apparently just returns null (api confusion :man_shrugging: )